onnx/backend/test/cpp/driver_test.cc

```diff
@@ -0,0 +1,138 @@
+#include "driver/gtest_utils.h"
+#include "onnx/checker.h"
+#include "onnx/onnxifi.h"
+#include "onnx/onnxifi_loader.h"
+const bool ONNXIFI_BACKEND_USED = false;
+class ONNXCppDriverTest
+	: public testing::TestWithParam<ProtoTestCase>{
+  protected:
+	  std::vector<ProtoTestData> protos_;
+	  onnx::ModelProto model_;
+	  void SetUp() override {
+		ProtoTestCase t = GetParam();
+		protos_ = t.proto_test_data_;
+		model_ = t.model_;
+	  }
+	  uint64_t getDescriptorSize(onnxTensorDescriptorV1 *t){
+		uint64_t d_size = 1;
+		for (int i = 0; i < t->dimensions; i++){
+			d_size *= t->shape[i];
+		}
+		return d_size;
+	  }
+	  void RunAndVerify(onnxifi_library& lib, onnxBackend& backend){
+            // Check Model
+            onnx::checker::check_model(model_);
+            // Check Input&Output Tensors
+            onnx::checker::CheckerContext ctx;
+            for (auto proto_test_data : protos_) {
+              for (auto input : proto_test_data.inputs_) {
+                onnx::checker::check_tensor(input, ctx);
+              }
+              for (auto output : proto_test_data.outputs_) {
+                onnx::checker::check_tensor(output, ctx);
+              }
+            }
+            /* TO DO:
+             * This chunk of code is to test the performance of onnxifi backend.
+             * Since we are not using a real backend, we should wait and not
+             * enable these tests. */
+
+            if (ONNXIFI_BACKEND_USED) {
+              onnxGraph graph;
+              uint32_t weightCount = model_.graph().initializer_size();
+              onnxTensorDescriptorV1 weightDescriptors =
+                  ProtoToOnnxTensorDescriptor(model_.graph().initializer(0));
```

Comment:
Why `initializer(0)`?

Reply:
since we need the address of the first initializer. Or we should put all initializers in a vector first?